air-script/tests/list_comprehension/list_comprehension.air

```diff
@@ -0,0 +1,29 @@
+def ListComprehensionAir
+
+trace_columns:
+    main: [clk, fmp[2], ctx]
+    aux: [a, b[3], c[4], d[4]]
+
+public_inputs:
+    stack_inputs: [16]
+
+boundary_constraints:
+    enf c[2].first = 0
+
+integrity_constraints:
+    let x = [fmp for fmp in fmp]
+    enf clk = x[1]
+    let enumerate = [2^i * c for (i, c) in (0..4, c)]
+    enf clk = enumerate[3]
+    let diff0 = [x - y for (x, y) in (c, d)]
+    enf clk = diff0[0]
+    let diff1 = [x - y for (x, y) in (c[0..2], d[1..3])]
+    enf clk = diff1[1]
+    let x0 = [w + x - y - z for (w, x, y, z) in (0..3, b, c[0..3], d[0..3])]
+    enf a = x0[0] + x0[1] + x0[2]
+    let x1 = sum([c * d for (c, d) in (c, d)])
+    let y1 = prod([c + d for (c, d) in (c, d)])    
+    enf clk = x1
+    let x2 = sum([c * d for (c, d) in (c, d)])
+    let y2 = [m + x2 for m in fmp]
+    enf clk = y2[0]
```

Comment:
minor nit: I'd probably add blank lines after each `enf` - would make it a bit easier to read.

Also, could we add a test for using "next" value from a column?